authentication/MultiAuthentication/MultiAuthenticationExtensions.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+using MultiAuthentication.Controllers;
+using MultiAuthentication.Handlers;
+using System.Reflection;
+using System.Security.Claims;
+
+namespace MultiAuthentication;
+
+public static class MultiAuthenticationExtensions
+{
+    public static void AddAuthenticationWithSchemeSelector(this IServiceCollection source)
+    {
+        source.AddMvcCore().ConfigureApplicationPartManager(manager =>
+        {
+            manager.ApplicationParts.Clear();
+            manager.ApplicationParts.Add(new SingleControllerApplicationPart<SchemeSelectorController>());
+        });
+
+        source.AddAuthentication(options =>
+        {
+            options.DefaultScheme = "ApiKey";
+            options.DefaultAuthenticateScheme = "ApiKey";
+            options.AddScheme<ApiKeyAuthenticationHandler>("ApiKey", "ApiKey");
+            options.AddScheme<BearerTokenAuthenticationHandler>("BearerToken", "BearerToken");
+            options.AddScheme<OrganizationIdAuthenticationHandler>("OrganizationId", "OrganizationId");
```

Comment:
rename these schemes to have abstract names `X`, `Y`, `Z`

Reply:
renamed the schemes to "Default", "Alternative", and "Anonymous"